Code/src/main/java/nl/utwente/group10/ui/components/FunctionBlock.java

```diff
@@ -23,17 +28,20 @@
 	private String[] arguments;
 	/** The name of this Function.**/
 	private String functionName;
-	
+	/** intstance to create Events for this FunctionBlock. **/
+	private static CustomGesture cg;
+		
 	/**
 	 * Method that creates a newInstance of this class along with it's visual representation
 	 * @param the number of arguments this FunctionBlock can hold
 	 * @return a new instance of this class
 	 * @throws IOException
 	 */
-	public static FunctionBlock newInstance(int numberOfArguments) throws IOException {
+	public static FunctionBlock newInstance(int numberOfArguments, CustomUIPane pane) throws IOException {
```

Comment:
Nicely spotted!

I don't think just taking the parent would be enough, though. At the very least, and I might be wrong in this, I think you'd have to walk the tree upwards, as I believe nodes (such as function blocks) can theoretically be nested. On the other hand, that case is not supported with the current approach either, which requires all blocks to be direct descendants of the pane.

To make a long story short: if all blocks are and will always remain direct descendants of Pane, go with @TheMSB's suggestion; if not, have Blocks (or FunctionBlocks) keep a reference to some global state like you do here, and perhaps add a method to find everything that looks like a block.

Alternatively, and this is the solution I like most even though it is probably a bit more work, have some observable global state object (perhaps Main?) [reference](https://docs.oracle.com/javafx/2/api/javafx/beans/value/ObservableObjectValue.html) a single Block that is *the* selected block, then push out events to every Block when that reference changes. Blocks would then each check if they're the selected block, and update their own state accordingly. This ensures that there is always just one (or always 0 or 1 if you wrap it in an Optional) object selected and keeps mutable state as simple as possible.